drivers/timer/xtensa_sys_timer.c

```diff
@@ -99,14 +101,175 @@ static u32_t __noinit cycles_per_tick;
 static u32_t __noinit max_system_ticks;
 static u32_t idle_original_ticks;
 static u32_t __noinit max_load_value;
+
+#ifdef CONFIG_TICKLESS_KERNEL
+static u32_t last_timer_value;
+#else
 static unsigned char timer_mode = TIMER_MODE_PERIODIC;
 static unsigned char idle_mode = IDLE_NOT_TICKLESS;
+#endif	/* CONFIG_TICKLESS_KERNEL */
+
+
+#ifdef CONFIG_TICKLESS_KERNEL
+
+/**
+ * @brief provides total programmed.
+ *
+ * returns : total number of sys ticks programmed.
```

Comment:
This isn't proper Doxygen. You shouldn't document these here anyway. This is part of the interface, should be documented in some header, if anywhere.